node-notes-cli/app.js

```diff
@@ -0,0 +1,60 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+// define operations
+const getOperation = { read, create, delete: deleteEntry, update };
+// try catch phases
+
+try {
+  let data = await readFile('data.json', 'utf-8');
+  data = JSON.parse(data);
+  const operation = process.argv[2];
+  if (Object.keys(getOperation).includes(operation)) {
+    getOperation[operation](data);
+  } else {
+    throw new Error(
+      `Operation ${operation} is not defined. Try create, read, update, delete only`
+    );
+  }
+} catch (err) {
+  console.log('Error: ', err);
+  process.exit(1);
+}
+
+// operation functions
+function read(data) {
+  for (const entryId in data.notes) {
+    console.log(`${entryId}: ${data.notes[entryId]}`);
+  }
+}
+
+async function create(data) {
+  const text = process.argv[3];
+  data.nextId++;
+  const id = data.nextId.toString();
+  data.notes[id] = text;
+  console.log(`Entry ${id} has been successfully created`);
+  await writeFile('data.json', JSON.stringify(data, null, 2), 'utf-8');
```

Comment:
You have this same line of code in here several times. It would be a good idea to extract it into a function.